src/Servers/Kestrel/Core/src/Features/IWebTransportSession.cs

```diff
@@ -0,0 +1,37 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Features;
+
+/// <summary>
+/// Controls the session and streams of a WebTransport session.
+/// </summary>
+public interface IWebTransportSession
+{
+    /// <summary>
+    /// The id of the WebTransport session.
+    /// </summary>
+    public long SessionId { get; }
+
+    /// <summary>
+    /// Abruptly close the WebTransport session and stop all the streams.
+    /// </summary>
+    /// <exception cref="Exception">If This is not a valid WebTransport session.</exception>
+    public void Abort();
```

Comment:
Should Abort take an int error code that will be sent to the client on the Connect stream RST? Maybe optional and we have a default.

Reply:
If the user calls HttpContext.Abort() it should trigger the same session cleanup.